### PR TITLE
fix: istioctl operator remove fail when not specify reversion

### LINKED ISSUE #43659: istioctl operator remove fail when not specify reversion
### Bug Description

When I use istioctl remove blow operator :

```shell
$ kubectl get po
NAME                             READY   STATUS    RESTARTS   AGE
istio-operator-64ccbfddc-m2wrh   1/1     Running   0          28s
$
$ istioctl operator remove
Operator controller is not installed in istio-operator namespace (no Deployment detected).
Aborting, use --force to override.
```

Debug:
```shell
$ istioctl operator remove --vklog=6
2023-02-28T15:23:09.883006Z	debug	klog	Config loaded from file:  /root/.kube/config
......
2023-02-28T15:23:09.916622Z	info	klog	GET https://127.0.0.1:6443/version?timeout=5s 200 OK in 1 milliseconds
2023-02-28T15:23:09.919840Z	info	klog	GET https://127.0.0.1:6443/apis/apps/v1/namespaces/istio-operator/deployments/istio-operator-default 404 Not Found in 2 milliseconds
Operator controller is not installed in istio-operator namespace (no Deployment detected).
```

The required deployment name is `isito-operator-default`, and we expect it to be `isito-operator`

### Version

```prose
istioctl version: Latest code of master branch
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by my-git9**
**Please provide a description of this PR:**
Fix istioctl operator remove fail when not specify reversion

Fix https://github.com/istio/istio/issues/43659

### CODE REVIEW

**Review by hanxiaop — CHANGES_REQUESTED** [review]
I think actually I fixed this before, but however due to some merge issues the fix was reverted.

File: operator/cmd/mesh/operator-common.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ const (
 // isControllerInstalled reports whether an operator deployment exists in the given namespace.
 func isControllerInstalled(cs kubernetes.Interface, operatorNamespace string, revision string) (bool, error) {
 	orn := operatorResourceName
-	if revision != "" {
+	if revision != "" && revision != "default" {
```
Comment (hanxiaop) [inline operator/cmd/mesh/operator-common.go:58]: I don't think it's correct to do it here. https://github.com/istio/istio/blob/ae5fb1a659b3876ae66f1a0a790591ea0155a7f6/operator/cmd/mesh/operator-remove.go#L86-L88 These lines revise the empty revision to "default", which should be incorrect.

Comment (my-git9) [reply operator/cmd/mesh/operator-common.go]: Thanks. I made adjustments according to your commit：https://github.com/istio/istio/commit/0c8164f2dc0f7f405fd6036614ca8c11a4452f7e

File: releasenotes/notes/43660.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
```
Comment (hanxiaop) [inline releasenotes/notes/43660.yaml:1]: I think we don't need a releasenote here. I wrote one before for the release: https://github.com/istio/istio/commit/0c8164f2dc0f7f405fd6036614ca8c11a4452f7e#diff-1208cdbb4e0a5998d314ac560f196977d73c184e22b9344ec4530410390fa09f, which resolved the same problem.

**Review by hanxiaop — APPROVED** [review]

**Comment by ericvn** [comment]
Can we get a release note for this, especially since there is an issue. Thanks.

**Comment by ericvn** [comment]
Can we get a release note for this, especially since there is an issue. Thanks.

**Comment by my-git9** [comment]
> Can we get a release note for this, especially since there is an issue. Thanks.
Thank you for your review, there is a related release note in the previous repair PR, you can check the discussion above:
https://github.com/istio/istio/pull/43660#discussion_r1121070073
Do you think we should add this release note again？

**Comment by my-git9** [comment]
> Can we get a release note for this, especially since there is an issue. Thanks.

Thank you for your review, there is a related release note in the previous repair PR, you can check the discussion above:
https://github.com/istio/istio/pull/43660#discussion_r1121070073
Do you think we should add this release note again？

**Comment by ericvn** [comment]
>  Do you think we should add this release note again？

Can we tweak the existing release note to include the reporting issue?

**Comment by my-git9** [comment]
> > Do you think we should add this release note again？
> 
> Can we tweak the existing release note to include the reporting issue?

Updated.

**Review by zirain — APPROVED** [review]

**Pull request merged**